python/tchannel/tornado/tchannel.py

```diff
@@ -103,15 +107,24 @@ def __init__(self, name, hostport=None, process_name=None,
         )
 
         self.name = name
+        self._zipkin = zipkin
 
         # register event hooks
         self.event_emitter = EventEmitter()
         self.hooks = EventRegistrar(self.event_emitter)
+        self.hooks.register(ZipkinTraceHook(tchannel=self))
 
         if known_peers:
             for peer_hostport in known_peers:
                 self.peers.add(peer_hostport)
 
+    @property
+    def zipkin(self):
+        if hasattr(self._zipkin, '__call__'):
```

Comment:
http://stackoverflow.com/questions/624926/how-to-detect-whether-a-python-variable-is-a-function